Integration tests!

amrav opened the pull request:
This brings us closer to being able to deploy master automatically with confidence.

amrav committed:
Split docker-compose.yml into multiple files

This allows us to specify a base (docker-compose.yml), common
configuration for devel (docker-compose.override.yml), and production
specific configuration (docker-compose.prod.yml). When developing, you
can continue to use `docker-compose ...` commands without passing any
special options.

This is nice because:
- We don't need to spin up some services in devel (like backups)
- It will make it possible to reuse the common configuration for
integration tests, which will specify their own overrides to avoid
clashing with any devel volumes.

amrav committed:
Add integration test scaffolding

- Define integration test config override, which basically renames all
  the volumes to avoid clashing with any existing volumes.
- Add test setup, which starts all the services and installs mediawiki
  using the CLI installer.
- Add an example test, which loads the Main Page. More to come.
- Add integration tests to the Travis CI config.

hargup reviewed (changes requested):
Can you add a few lines of instructions in the README about how to use the different docker files for the production and debugging purposes. Possibly including links to relevant documentation.

Also, at the moment you are testing if `mediawiki` is firing up properly, which is great! After the PR is merged please add issues that other more comprehensive tests needs to be added.

amrav committed:
Update Readme

- Add info about docker-compose multiple config files
- Add more instructions for running locally

This is not yet in a state where it's easy to run locally. Future
patches will reduce the complexity, but this is the best we can do for
now.

amrav commented:
Created #14 to track adding more integration tests. Hopefully the readme is a bit more informative now.

hargup committed:
Modify install link; add note about sudo

hargup committed:
fix typo
